### ISSUE TITLE
REPLACE should give error if variable is not character like

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1736]
in this case, pull_number is an integer,

```abap
DATA lv_uri TYPE string.
lv_uri = '/repos/{owner}/{repo}/pulls/{pull_number}sdfds'.
DATA(pull_number) = 1.
REPLACE ALL OCCURRENCES OF '{pull_number}' IN lv_uri WITH pull_number.
BREAK-POINT.
```
